apps/wallet-dashboard/components/Dialogs/SendToken/SendTokenDialog.tsx

```diff
@@ -56,14 +59,22 @@ function SendTokenDialogBody({
             addNotification('There was an error with the transaction', NotificationType.Error);
             return;
         } else {
+            setIsReviewing(true);
             signAndExecuteTransaction({
                 transaction,
             })
-                .then(() => {
-                    setOpen(false);
+                .then((d) =>
+                    iotaClient.waitForTransaction({
+                        digest: d.digest,
+                    }),
+                )
+                .then((transaction) => {
+                    setDigest(transaction.digest);
+                    setStep(FormStep.SentSuccess);
                     addNotification('Transfer transaction has been sent');
                 })
-                .catch(handleTransactionError);
+                .catch(handleTransactionError)
+                .finally(() => setIsReviewing(false));
```

Comment:
Once you have waited for `waitForTransaction`, there is a white screen until `SentSuccess` view is displayed. I think it's because once the first `then` finishes, it exits by the `finally`, leaving `setIsReviewing` to `false` but we continue at step `ReviewValues`
We have to improve this logic so that there is no white screen